osquery/tables/networking/darwin/wifi.mm

```diff
@@ -147,15 +146,23 @@ QueryData genKnownWifiNetworks(QueryContext& context) {
   }
 
   QueryData results;
-  auto count = CFDictionaryGetCount(networks);
-  const void* keys[count];
-  const void* values[count];
-  CFDictionaryGetKeysAndValues(networks, keys, values);
-
-  for (CFIndex i = 0; i < count; i++) {
-    parseNetworks((CFDictionaryRef)values[i], results);
+  if (CFGetTypeID(networks) == CFArrayGetTypeID()) {
+    auto count = CFArrayGetCount((CFArrayRef)networks);
+    for (CFIndex i = 0; i < count; i++) {
+      parseNetworks(
+          (CFDictionaryRef)CFArrayGetValueAtIndex((CFArrayRef)networks, i),
+          results);
+    }
+  } else if (CFGetTypeID(networks) == CFDictionaryGetTypeID()) {
+    auto count = CFDictionaryGetCount((CFDictionaryRef)networks);
+    const void* keys[count];
+    const void* values[count];
+    CFDictionaryGetKeysAndValues((CFDictionaryRef)networks, keys, values);
+
+    for (CFIndex i = 0; i < count; i++) {
+      parseNetworks((CFDictionaryRef)values[i], results);
```

Comment:
I'm wondering-- if this `values[1]`, or the above `networks`, contained something other than a dict-- if the `parseNetworks` code would barf? Might need to make use of some more Apple RTTI and do a paranoid check that these are `CFDictionaryGetTypeID` types. I think you have a guarantee that it's either `nullptr` or some derived `CFTypeRef`.